superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -247,6 +287,16 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
         // don't pass parameters if using the sqlalchemy uri
         delete update.parameters;
       }
+
+      // // Structure extra_json
+      // const extraValuesStructured = {
+      //   ...extraValues,
+      //   ...db?.extra_json,
+      // };
+
+      // Add values back to extra field
+      update.extra = JSON.stringify(db?.extra_json, null, '  ');
```

Comment:
what about the idea to store this in the `db` object as a string and then parse it out in the extraOptions component? So you don't have to manipulate anything here.